ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-main.yaml

```diff
@@ -4,6 +4,70 @@ build_root:
     name: release
     namespace: openshift
     tag: golang-1.17
+images:
+- build_args:
+  - name: ARCH
+    value: amd64
+  - name: ldflags
+  dockerfile_literal: |-
+    FROM registry.ci.openshift.org/openshift/release:golang-1.17 as toolchain
+
+    # Run this with docker build --build_arg $(go env GOPROXY) to override the goproxy
+    ARG goproxy=https://proxy.golang.org
+    ENV GOPROXY=$goproxy
+    ENV GOFLAGS="-mod=readonly"
+
+    FROM toolchain as builder
+    WORKDIR /workspace
+
+    # Copy the Go Modules manifests
+    COPY go.mod go.mod
+    COPY go.sum go.sum
+    # Cache deps before building and copying source so that we don't need to re-download as much
+    # and so that source changes don't invalidate our downloaded layer
+    RUN  --mount=type=cache,target=/root/.local/share/golang \
+    --mount=type=cache,target=/go/pkg/mod \
+    go mod download
+
+    # Copy the sources
+    COPY ./ ./
+
+    # Build
+    ARG package=.
+    ARG ARCH
+    ARG LDFLAGS
+    RUN --mount=type=cache,target=/root/.cache/go-build \
+    --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target=/root/.local/share/golang \
+    CGO_ENABLED=0 GOOS=linux GOARCH=${ARCH} go build -ldflags "${LDFLAGS} -extldflags '-static'"  -o manager ${package}
+    ENTRYPOINT [ "/start.sh", "/workspace/manager" ]
+
+    # Copy the controller-manager into a thin image
+    FROM gcr.io/distroless/static:nonroot
+    WORKDIR /
+    COPY --from=builder /workspace/manager .
+    # Use uid of nonroot user (65532) because kubernetes expects numeric user when applying pod security policies
+    USER 65532
+    ENTRYPOINT ["/manager"]
+  from: base
+  inputs:
+    openshift_release_golang-1.17:
+      as:
+      - registry.ci.openshift.org/openshift/release:golang-1.17
+  to: cluster-api-kubevirt-controller
```

Comment:
for consistency with other openshift forks, i think this should be called `kubevirt-machine-controllers"

Reply:
I copied it from AWS repo

Reply:
@davidvossel the `aws-machine-controller` term is used for the code and images used for `openshift/machine-api-operator`
capk code is not used for `openshift/machine-operator` therefore we shouldn't use the same term
Also this image shouldn't be added to the release image
Therefore I agree that currently we should build the image the same way it done in `capa` for hypershift
This is the relevant PR: https://github.com/openshift/release/pull/23913
Please notice that in `openshift/cluster-api-provider-aws` there are two different images:
1. The image used for `openshift/machine-api-operator` based on the code from the master
2. The image for hypershift capa - based on code from upstream, special branches are created from upstream

Reply:
got it, makes sense. I see what you're saying is consistent with the aws controllers consumed by hypershift. thanks for the explanation

```
        // Image built from https://github.com/openshift/cluster-api-provider-aws/tree/release-1.1
        // Upstream canonical image comes from  https://console.cloud.google.com/gcr/images/k8s-artifacts-prod
        // us.gcr.io/k8s-artifacts-prod/cluster-api-aws/cluster-api-aws-controller:v1.1.0
        imageCAPA = "registry.ci.openshift.org/hypershift/cluster-api-aws-controller:v1.1.0"

```